servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusNessieAuthorizationConfig.java

```diff
@@ -15,29 +15,28 @@
  */
 package org.projectnessie.server.config;
 
-import io.smallrye.config.ConfigMapping;
-import io.smallrye.config.WithDefault;
-import io.smallrye.config.WithName;
-import java.util.Map;
+import io.quarkus.arc.config.ConfigProperties;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 
 /** Configuration for Nessie authorization settings. */
-@ConfigMapping(prefix = "nessie.server.authorization")
+@ConfigProperties(prefix = "nessie.server.authorization")
```

Comment:
TBH we shouldn't roll back these changes. Then we also wouldn't need the workaround that was added in `CompiledAuthorizationRules`.

Also in fact I believe we only need to roll back the changes in `QuarkusServerConfig`. Everything else can use the new `ConfigMapping`

Reply:
> only need to roll back the changes in QuarkusServerConfig

Everything that's `@Inject`ed via a constructor is affected by the Quarkus isssue.

Reply:
as far as I remember, the problem was always the injection of `ServerConfig` into `NessieExceptionMapper`. The changes from https://github.com/projectnessie/nessie/pull/1967 work for me locally. Let's see if native tests pass with that minimal set of changes